pkg/keycloak/keycloak.go

```diff
@@ -382,11 +384,58 @@ func (h *Handler) reconcileRealms(kc *v1alpha1.Keycloak, kcClient KeycloakInterf
 
 	for name, realmPair := range realmPairsList {
 		err = h.reconcileRealm(realmPair.KcRealm, realmPair.ObjRealm, kcClient)
-		//This should try and reconcile all realms rather throwing an error on the first failure
+		// This should try and reconcile all realms rather throwing an error on the first failure
 		if err != nil {
 			return errors.Wrapf(err, "error reconciling realm: %v", name)
 		}
 	}
+
+	err = h.reconcileClients(kc, kcClient, kcRealmMap)
+	if err != nil {
+		return errors.Wrap(err, "error reconciling clients")
+	}
```

Comment:
I'm coming round to thinking that we should just add the reconcile<realm sub resource> into the reconcileRealm method and only call them when we haven't just created the realm. This also leaves whether we are acting upon a realm up to the logic in reconcileRealm rather than in the method for every sub resource i.e. You wouldn't need all these checks "realmName != "master""